deploy/cloud/operator/internal/dynamo/graph.go

```
@@ -527,3 +541,28 @@ func mergeEnvs(common, specific []corev1.EnvVar) []corev1.EnvVar {
 	}
 	return merged
 }
+
+func overrideExtraPodSpecMainContainer(src *common.ExtraPodSpec, dst *common.ExtraPodSpec) {
+	if src == nil || src.MainContainer == nil {
+		return
+	}
+
+	commandIsSet := len(src.MainContainer.Command) > 0
+	argsIsSet := len(src.MainContainer.Args) > 0
+
+	if !commandIsSet && !argsIsSet {
+		return
+	}
+
+	if dst == nil {
+		panic("overrideExtraPodSpecMainContainer: dst must not be nil")
```

never panic, you should return an error instead.
if `deployment.Spec.DynamoComponentDeploymentSharedSpec.ExtraPodSpec` is nil, we should initialize it instead in this case